cmd/optimize/main.go

```diff
@@ -0,0 +1,682 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/csv"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/log"
+	"github.com/containerd/containerd/pkg/netns"
+	"github.com/containerd/containerd/platforms"
+	gocni "github.com/containerd/go-cni"
+	"github.com/containerd/stargz-snapshotter/analyzer"
+	stargzcommands "github.com/containerd/stargz-snapshotter/cmd/ctr-remote/commands"
+	"github.com/containerd/stargz-snapshotter/recorder"
+	"github.com/containerd/stargz-snapshotter/util/containerdutil"
+	"github.com/hashicorp/go-multierror"
+	imagespec "github.com/opencontainers/image-spec/specs-go/v1"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	runtimespec "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/rs/xid"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
+
+	"github.com/containerd/containerd/cmd/ctr/commands"
+	"github.com/containerd/containerd/containers"
+	"github.com/containerd/containerd/content"
+	"github.com/containerd/containerd/contrib/nvidia"
+	"github.com/containerd/containerd/images"
+	"github.com/containerd/containerd/oci"
+)
+
+const defaultPeriod = 10
+const netnsMountDir = "/var/run/netns"
+
+const DefaultAddress = "/run/containerd/containerd.sock"
+const (
+	// NamespaceEnvVar is the environment variable key name
+	NamespaceEnvVar = "CONTAINERD_NAMESPACE"
+	// Default is the name of the default namespace
+	DefaultNamespace = "default"
+)
+
+func main() {
+	app := cli.NewApp()
+	app.Flags = containerdFlags
+	app.Name = "optimize"
+	app.Usage = "Generate accessed files list for OCI format image."
```

Comment:
Thanks for adding this tool, maybe give some docs and mark the subcommand `experimental`?